aws-controllers-k8s/rds-controller
# Add reference for db instance/cluster param/subnet/vpcsecurity group

Issue #, if available:
https://github.com/aws-controllers-k8s/community/issues/1378

Description of changes:
Add reference for DBInstance/Cluster ParameterGroup/SubnetGroup/vpcSecurityGroup. 

Here is my testing process, it should apply to param group and vpc security group too. 

1. Try create a rds instance `brucegu-test-ref-4` and specify 
```  
dbSubnetGroupRef:
    from:
      name: "brucegu-test-subnet-11""
```
 but no `brucegu-test-subnet-11` is available yet. 
2. Verify db instance state has 
```
   Message:               dbsubnetgroups.rds.services.k8s.aws "brucegu-test-subnet-11" not found
    Status:                Unknown
    Type:                  ACK.ReferencesResolved
```
3. Create db SubnetGroup  `brucegu-test-subnet-11` from SubnetGroup CRD 
4. Verify db instance `brucegu-test-ref-4` is created successfully and Resource synced successfully.
```
  Conditions:
    Last Transition Time:     2022-07-07T22:09:25Z
    Status:                   True
    Type:                     ACK.ReferencesResolved
    Last Transition Time:     2022-07-07T22:09:26Z
    Message:                  Late initialization successful
    Reason:                   Late initialization successful
    Status:                   True
    Type:                     ACK.LateInitialized
    Last Transition Time:     2022-07-07T22:09:26Z
    Message:                  Resource synced successfully
    Reason:
    Status:                   True
    Type:                     ACK.ResourceSynced
```


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
